api.go

```diff
@@ -138,31 +138,39 @@ func HandleQueryValidation(c *gin.Context) {
 	absPath, _ := filepath.Abs("../searchrefiner/dictionary/words.txt")
 	keywordDictionary := loader.LoadDictionary(absPath)
 
+	lang = strings.ToLower(lang)
+
 	var fieldsError []string
 
-	if lang == "medline" {
+	if strings.ToLower(lang) == "medline" {
 		scanner := bufio.NewScanner(strings.NewReader(rawQuery))
 		var extractedFields []string
 		for scanner.Scan() {
 			temp := scanner.Text()
 			line := temp[3:]
 			reg := regexp.MustCompile(`\.([^.]+)\.`)
 			rawFields := reg.FindAllStringSubmatch(line, -1)
-			extractedFields = append(extractedFields, rawFields[0][1])
+			if len(rawFields) > 0 {
+				extractedFields = append(extractedFields, rawFields[0][1])
+			}
 		}
-		for _, i := range extractedFields {
-			flag := checker.CheckWord(fieldsDictionary, strings.ToLower(i), 0)
-			if !flag {
-				fieldsError = append(fieldsError, i)
+		if len(extractedFields) > 0 {
```

Comment:
This check doesn't do anything, since Go will not range over an empty slice.

Reply:
Actually, the `extractedFields` can be `[]` if there is no fields presented in the query, like for Medline, `1 or 2` or for Pubmed, `brain and cancer`.